pkg/media/ivf-writer_test.go

```diff
@@ -0,0 +1,151 @@
+package media
+
+import (
+	"fmt"
+	"os"
+	"testing"
+
+	"github.com/pions/rtp"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestIVFWriter_FileCreation(t *testing.T) {
+	assert := assert.New(t)
+
+	// Create a IVFWriter with an empty filename
+	writer, err := NewIVFWriter("")
+	assert.NotNil(err, "NewIVFWriter shouldn't create file with an empty name")
+	assert.Nil(writer, "NewIVFWriter should return a nil value for the writer, in case of error")
+
+	// Creates a valid file, and tries to open it twice
+	writer, err = NewIVFWriter("/tmp/pions_testCreateIVFWriter1.ivf")
+	assert.Nil(err, "NewIVFWriter should create file")
+	assert.NotNil(writer, "NewIVFWriter shouldn't return a nil value for the writer, in case of success")
+	err = writer.open("/tmp/pions_testCreateIVFWriter1.ivf")
+	assert.NotNil(err, "IVFWriter shouldn't be able to call open() twice")
+	err = os.Remove("/tmp/pions_testCreateIVFWriter1.ivf")
+	assert.Nil(err, "File should be removable")
+}
+
+type ivfWriterPacketTest struct {
+	fileName     string
+	message      string
+	messageClose string
+	packet       *rtp.Packet
+	writer       Writer
+	err          error
+	closeErr     error
+}
+
+func TestIVFWriter_AddPacketAndClose(t *testing.T) {
+	assert := assert.New(t)
+
+	// The linter misbehave and thinks this code is the same as the tests in opus-writer_test
+	// nolint:dupl
+	addPacketTestCase := []ivfWriterPacketTest{
+		{
+			fileName:     "/tmp/pions_testIVFPacket1.ivf",
+			message:      "IVFWriter shouldn't be able to write something to a closed file",
+			messageClose: "IVFWriter should be able to close an already closed file",
+			packet:       nil,
+			err:          fmt.Errorf("file not opened"),
+			closeErr:     nil,
+		},
+		{
+			fileName:     "/tmp/pions_testIVFPacket2.ivf",
+			message:      "IVFWriter shouldn't be able to write something else than an IVF packet",
+			messageClose: "IVFWriter should be able to close the file",
+			packet:       &rtp.Packet{},
+			err:          fmt.Errorf("Payload is not large enough to container header"),
+			closeErr:     nil,
+		},
+		{
+			fileName:     "/tmp/pions_testIVFPacket3.ivf",
+			message:      "IVFWriter shouldn't be able to write something else than an IVF packet",
+			messageClose: "IVFWriter should be able to close the file",
+			packet: &rtp.Packet{
+				Header: rtp.Header{
+					Marker:           true,
+					Extension:        true,
+					ExtensionProfile: 1,
+					ExtensionPayload: []byte{0xFF, 0xFF, 0xFF, 0xFF},
+					Version:          2,
+					PayloadOffset:    20,
+					PayloadType:      defaultPayloadTypeOpus,
+					SequenceNumber:   27023,
+					Timestamp:        3653407706,
+					SSRC:             476325762,
+					CSRC:             []uint32{},
+				},
+				Payload: rawPkt[20:],
+				Raw:     rawPkt,
+			},
+			err:      nil, // TODO: Update expected value
```

Comment:
Currently, when unmarshaling a packet, `pions/rtp` doesn't check the `PayloadType`, so it doesn't fail (it should IMHO)